src/honeyhive/tracer/__init__.py

```diff
@@ -552,3 +552,246 @@ def enrich_session(
+        '123e4567-e89b-12d3-a456-426614174000'
+    """
+    try:
+        uuid_obj = uuid.UUID(guid)
+        # Check if it's version 4
+        if uuid_obj.version != 4:
+            raise ValueError("UUID must be version 4")
+        # Return lowercase string
+        return str(uuid_obj).lower()
+    except ValueError as e:
+        raise ValueError(f"Invalid UUIDv4: {str(e)}. Please provide a valid UUIDv4.")
+
+def start_session(
+    api_key: str = None,
+    project: str = None,
+    source: str = None,
+    config: dict = None,
+    inputs: dict = None,
+    metadata: dict = None,
+    user_properties: dict = None,
+    session_id: str = None,
+    capture_git_info: bool = False,
+    session_name: str = "unknown",
+    server_url: str = 'https://api.honeyhive.ai',
+    verbose: bool = False
+) -> str:
+    """
+    Start a new session with HoneyHive.
+    
+    Args:
+        api_key (str, optional): Your HoneyHive API key. Must be provided or set via HH_API_KEY env var.
+        project (str, optional): The project name. Must be provided or set via HH_PROJECT env var.
+        session_name (str, optional): Name for this session. Defaults to script name.
+        source (str, optional): Source of the session. Defaults to "dev" or HH_SOURCE env var.
+        config (dict, optional): Configuration for the session.
+        inputs (dict, optional): Input parameters for the session.
+        metadata (dict, optional): Additional metadata for the session.
+        user_properties (dict, optional): User-defined properties for the session.
+        session_id (str, optional): A valid UUIDv4 for the session. If not provided, one will be generated.
+        server_url (str, optional): HoneyHive API server URL. Defaults to "https://api.honeyhive.ai" or HH_API_URL env var.
+        
+    Returns:
+        str: The session ID (UUIDv4)
+        
+    Raises:
+        Exception: If required parameters are missing or invalid
+        
+    Examples:
+        >>> start_session(api_key="your-api-key", project="my-project")
+        '123e4567-e89b-12d3-a456-426614174000'
+    """
+    try:
+        # Get server URL from env if not provided
+        if server_url == 'https://api.honeyhive.ai':
+            server_url = os.getenv("HH_API_URL", server_url)
+        
+        if api_key is None:
+            api_key = os.getenv("HH_API_KEY")
+            if api_key is None:
+                raise Exception("api_key must be specified or set in environment variable HH_API_KEY.")
+        
+        # Get project from env if not provided
+        if project is None:
+            project = os.getenv("HH_PROJECT")
+            if project is None:
+                raise Exception("project must be specified or set in environment variable HH_PROJECT.")
+        
+        # Get source from env if not provided
+        if source is None:
+            source = os.getenv("HH_SOURCE", "dev")   
+
+        if capture_git_info:
```

Comment:
can we re-use all this logic from the tracer init? https://github.com/honeyhiveai/python-sdk/blob/main/src/honeyhive/tracer/__init__.py#L71-L99